internal/verifier/change_stream.go

```diff
@@ -140,37 +144,46 @@ func (verifier *Verifier) iterateChangeStream(ctx context.Context, cs *mongo.Cha
 		return err
 	}
 
-	readAndHandleOneChangeEventBatch := func() (bool, error) {
+	readAndHandleOneChangeEventBatch := func() error {
 		eventsRead := 0
 		var changeEventBatch []ParsedEvent
 
 		for hasEventInBatch := true; hasEventInBatch; hasEventInBatch = cs.RemainingBatchLength() > 0 {
 			gotEvent := cs.TryNext(ctx)
 
-			if !gotEvent || cs.Err() != nil {
+			if cs.Err() != nil {
+				return errors.Wrap(cs.Err(), "change stream iteration failed")
+			}
+
+			if !gotEvent {
 				break
 			}
 
+			verifier.logger.Debug().
+				Stringer("resumeToken", cs.ResumeToken()).
+				Msg("TryNext got an event!")
+
 			if changeEventBatch == nil {
 				changeEventBatch = make([]ParsedEvent, cs.RemainingBatchLength()+1)
 			}
 
 			if err := cs.Decode(&changeEventBatch[eventsRead]); err != nil {
-				return false, errors.Wrap(err, "failed to decode change event")
+				return errors.Wrap(err, "failed to decode change event")
 			}
 
 			eventsRead++
 		}
 
-		if eventsRead > 0 {
-			verifier.logger.Debug().Int("eventsCount", eventsRead).Msgf("Received a batch of events.")
```

Comment:
(Just an observation that this log line has disappeared in the shuffle; I have no opinion on whether that's good or bad.)